src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -28145,6 +28145,92 @@ public C(string? p, [NotNullIfNotNull(""p"")] string? pOut)
                 Diagnostic(ErrorCode.WRN_ParameterNotNullIfNotNull, "}").WithArguments("pOut", "p").WithLocation(26, 5));
         }
 
+        [Fact, WorkItem(44539, "https://github.com/dotnet/roslyn/issues/44539")]
+        public void NotNullIfNotNull_Constructor_02()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    public C() { }
+
+    public C(string? p, [NotNullIfNotNull(""p"")] out string? pOut) : this()
+    {
+        if (p is object)
+        {
+            pOut = null;
+            return; // 1
+        }
+
+        if (p is object)
+        {
+            pOut = p;
+            return;
+        }
+
+        if (p is null)
+        {
+            pOut = null;
+            return;
+        }
+
+        pOut = M1();
+    } // 2
+
+    string? M1() => null;
+}
+";
+            var comp = CreateNullableCompilation(new[] { source, NotNullIfNotNullAttributeDefinition });
+            comp.VerifyDiagnostics(
+                // (12,13): warning CS8825: Parameter 'pOut' must have a non-null value when exiting because parameter 'p' is non-null.
+                //             return; // 1
+                Diagnostic(ErrorCode.WRN_ParameterNotNullIfNotNull, "return;").WithArguments("pOut", "p").WithLocation(12, 13),
+                // (28,5): warning CS8825: Parameter 'pOut' must have a non-null value when exiting because parameter 'p' is non-null.
+                //     } // 2
+                Diagnostic(ErrorCode.WRN_ParameterNotNullIfNotNull, "}").WithArguments("pOut", "p").WithLocation(28, 5));
+        }
+
+        [Fact, WorkItem(44539, "https://github.com/dotnet/roslyn/issues/44539")]
+        public void NotNullIfNotNull_Constructor_03()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    public C(string x, string? y) : this(x, out y)
+    {
+        y.ToString();
+    }
+
+    public C(string? p, [NotNullIfNotNull(""p"")] out string? pOut) { pOut = p; }
+}
+";
+            var comp = CreateNullableCompilation(new[] { source, NotNullIfNotNullAttributeDefinition });
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(44539, "https://github.com/dotnet/roslyn/issues/44539")]
+        public void NotNull_Constructor()
```

Comment:
This test uncovered a problem where the NotNull diagnostic would be given once in the "initializers" pass, and again in the "body" pass. Hence the implementation change.